### ISSUE TITLE
Support node 4.x and higher

### ISSUE THREAD

**Issue by shadeglare** [SheetJS/js-xlsx#369]
Some parts of code are not compatible with node 4.x and higher. Like inappropriate using the Buffer length property which is now readonly. Any possibilities that could be resolved?

**Comment by pietersv** [comment]
Aha. To confirm I just ran the (amazingly thorough) test suite and see some of the tests  of .xlsx failing on node 4.1 on a clean clone.
```
$> make test

1) should parse test files AutoFilter.xlsb [.xlsb]:
     Error: Unexpected record 0 BrtRowHdr
      at hopper_sty (/Users/me/Dev/js-xlsx/xlsx.js:4915:41)
      at recordhopper (/Users/pieter/Dev/js-xlsx/xlsx.js:1693:6)
      at parse_sty_bin (/Users/me/Dev/js-xlsx/xlsx.js:4856:2)
      at parse_sty (/Users/me/Dev/js-xlsx/xlsx.js:8436:67)
      at parse_zip (/Users/me/Dev/js-xlsx/xlsx.js:11176:26)
      at read_zip (/Users/me/Dev/js-xlsx/xlsx.js:11380:9)
      at Object.readSync [as read] (/Users/me/Dev/js-xlsx/xlsx.js:11396:11)
      at Context.<anonymous> (/Users/me/Dev/js-xlsx/test.js:193:20)
...
```

and if I comment out tests of xlsb files then get 

```
  1) should parse test files biff5/NumberFormatCondition.xls:
     TypeError: Cannot set property length of [object Object] which has only a getter
      at sbcs_d (/Users/me/Dev/js-xlsx/dist/cpexcel.js:909:20)
      at Object.decode (/Users/me/Dev/js-xlsx/dist/cpexcel.js:1197:40)
      at _gc2 (/Users/me/Dev/js-xlsx/xlsx.js:28:24)
      at Buffer.ReadShift [as read_shift] (/Users/me/Dev/js-xlsx/xlsx.js:1623:13)
      at parse_ShortXLUnicodeString (/Users/me/Dev/js-xlsx/xlsx.js:3082:21)
      at Object.parse_Font [as f] (/Users/me/Dev/js-xlsx/xlsx.js:3470:13)
      at slurp (/Users/me/Dev/js-xlsx/xlsx.js:9229:11)
      at parse_workbook (/Users/me/Dev/js-xlsx/xlsx.js:9350:15)
      at parse_xlscfb (/Users/me/Dev/js-xlsx/xlsx.js:9827:27)
      at readSync (/Users/me/Dev/js-xlsx/xlsx.js:11391:11)
     ...

```

**Comment by xxScorpion** [comment]
node_modules\xlsx\dist\cpexcel.js:909
        mdb.length = 2 * len;
                  ^
TypeError: Cannot set property length of [object Object] which has only a getter

To correct you need "mdb = new Buffer(mdl)" to change  "mdb = Array(mdl)"
Can anyone make corrections to the repository ?

**Comment by blakehagen** [comment]
It appears @fluciotto has a robust fix for this issue as opposed to just commenting out the line as others have. Is this something that you will try to merge into master or should I just use @fluciotto's fork?

**Comment by pietersv** [comment]
@blakehagen that's fantastic news. I'd prefer to merge, will have time next week.  If your work is more urgent, use the other fork, and we can connect in a bit.

**Comment by reviewher** [comment]
Fixed in 0.8.1, test suite currently runs against node versions 0.8 0.9 0.10 0.12 4.2 5 6 7

If you find any issues where something works in the latest iteration of one node version but not another, it is a bug and please open a new issue.
